p5/sketch/Vispy2DRenderer/base.py

```diff
@@ -93,32 +96,39 @@ def on_timer(self, event):
         self.measure_fps(callback=lambda _: None)
         builtins.frame_rate = round(self.fps, 2)
 
-        with p5.renderer.draw_loop():
-            if not self.setup_done:
-                builtins.frame_count += 1
-                self.setup_method()
-                self.setup_done = True
-                self.show(visible=True)
-                if self.redraw is None:
+        try:
+            with p5.renderer.draw_loop():
+                if not self.setup_done:
+                    builtins.frame_count += 1
+                    self.setup_method()
+                    self.setup_done = True
+                    self.show(visible=True)
+                    if self.redraw is None:
+                        self.redraw = False
+                    if self.looping is None:
+                        self.looping = True
+
+                elif self.redraw:
+                    builtins.frame_count += 1
+                    self.draw_method()
+                    self.redraw = False
+                elif self.looping:
+                    builtins.frame_count += 1
+                    self.draw_method()
                     self.redraw = False
-                if self.looping is None:
-                    self.looping = True
-
-            elif self.redraw:
-                builtins.frame_count += 1
-                self.draw_method()
-                self.redraw = False
-            elif self.looping:
-                builtins.frame_count += 1
-                self.draw_method()
-                self.redraw = False
-            elif not self.looping:
-                pass
-
-            while len(self.handler_queue) != 0:
-                function, event = self.handler_queue.pop(0)
-                event._update_builtins()
-                function(event)
+                elif not self.looping:
+                    pass
+
+                if self.exiting:
+                    raise ExitException()
```

Comment:
Why are we raising an exit exception here, instead of directly calling `self.close()` and returning.